### ISSUE TITLE
The zipline demo didn't work as expected.

### ISSUE THREAD

**Issue by Quevi** [stefan-jansen/machine-learning-for-trading#194]
**Describe the bug**
When I tried to run the zipline demo, I met the error: “Exception: Must provide either benchmark_asset or benchmark_returns.”

**To Reproduce**

1. Go to './02_market_and_fundamental_data/03_data_providers/05_zipline_data_demo.ipynb'
2. Keep running each cell until the first cell that starts with '%%zipline'
3. An error occured: 'Exception: Must provide either benchmark_asset or benchmark_returns.'

**Expected behavior**

It was expected to see a table as show in github-hosted notebook as below:
<img width="1186" alt="截屏2021-08-14 下午10 02 51" src="https://user-images.githubusercontent.com/22657849/129448777-8aa730bb-9f11-4b69-a9b5-38703239e0ff.png">

**Screenshots**
I saw this:
<img width="1123" alt="截屏2021-08-14 下午9 54 29" src="https://user-images.githubusercontent.com/22657849/129448568-5534db8c-6b2e-4a3a-a6c0-b486806a2b99.png">
The full infomation is:
```
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-6-087b07fb55a8> in <module>
----> 1 get_ipython().run_cell_magic('zipline', '--start 2010-1-1 --end 2018-1-1 --data-frequency daily', 'from zipline.api import order_target, record, symbol\nimport pandas as pd\n\ndef initialize(context):\n    context.i = 0\n    context.assets = [symbol(\'FB\'), symbol(\'GOOG\'), symbol(\'AMZN\')]\n    \ndef handle_data(context, data):\n    df = data.history(context.assets, fields=[\'price\', \'volume\'], bar_count=1, frequency="1d")\n    df = df.to_frame().reset_index()\n    \n    if context.i == 0:\n        df.columns = [\'date\', \'asset\', \'price\', \'volume\']\n        df.to_csv(\'stock_data.csv\', index=False)\n    else:\n        df.to_csv(\'stock_data.csv\', index=False, mode=\'a\', header=None)\n    context.i += 1\n')

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/IPython/core/interactiveshell.py in run_cell_magic(self, magic_name, line, cell)
   2397             with self.builtin_trap:
   2398                 args = (magic_arg_s, cell)
-> 2399                 result = fn(*args, **kwargs)
   2400             return result
   2401 

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/zipline/__main__.py in zipline_magic(line, cell)
    333     )
    334     try:
--> 335         return run.main(
    336             # put our overrides at the start of the parameter list so that
    337             # users may pass values with higher precedence

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/click/core.py in main(self, args, prog_name, complete_var, standalone_mode, **extra)
    780             try:
    781                 with self.make_context(prog_name, args, **extra) as ctx:
--> 782                     rv = self.invoke(ctx)
    783                     if not standalone_mode:
    784                         return rv

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/click/core.py in invoke(self, ctx)
   1064         _maybe_show_deprecated_notice(self)
   1065         if self.callback is not None:
-> 1066             return ctx.invoke(self.callback, **ctx.params)
   1067 
   1068 

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/click/core.py in invoke(*args, **kwargs)
    608         with augment_usage_errors(self):
    609             with ctx:
--> 610                 return callback(*args, **kwargs)
    611 
    612     def forward(*args, **kwargs):  # noqa: B902

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/click/decorators.py in new_func(*args, **kwargs)
     19 
     20     def new_func(*args, **kwargs):
---> 21         return f(get_current_context(), *args, **kwargs)
     22 
     23     return update_wrapper(new_func, f)

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/zipline/__main__.py in run(ctx, algofile, algotext, define, data_frequency, capital_base, bundle, bundle_timestamp, benchmark_file, benchmark_symbol, benchmark_sid, no_benchmark, start, end, output, trading_calendar, print_algo, metrics_set, local_namespace, blotter)
    298     )
    299 
--> 300     return _run(
    301         initialize=None,
    302         handle_data=None,

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/zipline/utils/run_algo.py in _run(handle_data, initialize, before_trading_start, analyze, algofile, algotext, defines, data_frequency, capital_base, bundle, bundle_timestamp, start, end, output, trading_calendar, print_algo, metrics_set, local_namespace, environ, blotter, custom_loader, benchmark_spec)
    194 
    195     try:
--> 196         perf = TradingAlgorithm(
    197             namespace=namespace,
    198             data_portal=data,

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/zipline/algorithm.py in run(self, data_portal)
    635         try:
    636             perfs = []
--> 637             for perf in self.get_generator():
    638                 perfs.append(perf)
    639 

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/zipline/algorithm.py in get_generator(self)
    611         method to get a standard construction generator.
    612         """
--> 613         return self._create_generator(self.sim_params)
    614 
    615     def run(self, data_portal=None):

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/zipline/algorithm.py in _create_generator(self, sim_params)
    572             self.initialized = True
    573 
--> 574         benchmark_source = self._create_benchmark_source()
    575 
    576         self.trading_client = AlgorithmSimulator(

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/zipline/algorithm.py in _create_benchmark_source(self)
    538             benchmark_asset = None
    539             benchmark_returns = self.benchmark_returns
--> 540         return BenchmarkSource(
    541             benchmark_asset=benchmark_asset,
    542             benchmark_returns=benchmark_returns,

~/miniconda3/envs/ml4t/lib/python3.8/site-packages/zipline/sources/benchmark_source.py in __init__(self, benchmark_asset, trading_calendar, sessions, data_portal, emission_rate, benchmark_returns)
     66                 self._precalculated_series = daily_series
     67         else:
---> 68             raise Exception(
     69                 "Must provide either benchmark_asset or " "benchmark_returns."
     70             )

Exception: Must provide either benchmark_asset or benchmark_returns.
```

**Environment**
 
- OS: MacOSX
 - Version: Big Sur 11.3.1

**Additional context**
1. My machine is macbookpro 2020 with M1 chip. 
2. I git cloned the latest version on August 14th, 2021.
2. I followed every step in ./installation/README.md, creating a virtual environment with mamba and ./installation/macosx/ml4t.yml

**Comment by Quevi** [comment]
It seems to work to add '--no-benchmark' to the zipline magic fuction (and also some compatibility problems with pandas Dataframe need to be fixed).

**Comment by stefan-jansen** [comment]
Fixed by #198 .
